packages/design-system/src/components/dropdown-menu/dropdown-menu.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  ButtonHTMLAttributes,
+  createContext,
+  ReactNode,
+  useContext,
+  useRef,
+  useState,
+} from 'react';
+
+import { cn } from '../../utils';
+import useClickOutside from '../../utils/use-click-outside';
+
+import * as styles from './dropdown-menu.css';
+
+interface DropdownContextProps {
+  open: boolean;
+  handleToggleOpen: () => void;
+  handleToggleClose: () => void;
+}
+
+const DropdownContext = createContext<DropdownContextProps | null>(null);
+
+// useDropdownContext: 드롭다운 컴포넌트 외부에서 서브 컴포넌트들이 사용됐을 때 에러 처리
+const useDropdownContext = () => {
+  const context = useContext(DropdownContext);
+  if (!context) {
+    throw new Error('Dropdown 컴포넌트는 Dropdown 내에서 사용되어야 합니다.');
+  }
+  return context;
+};
+
+interface DropdownRootProps {
+  children: ReactNode;
+  className?: string;
+}
+
+const DropdownRoot = ({ className, children }: DropdownRootProps) => {
+  const [open, setOpen] = useState(false);
+
+  const handleToggleOpen = () => setOpen((prev) => !prev);
+  const handleClose = () => setOpen(false);
+
+  const ref = useRef(null);
```

Comment:
```suggestion
  const ref = useRef<HTMLInputElement>(null);
```
DOM을 조작하기 위한 ref는 해당 방식으로 타입을 지정해주는게 더 좋을듯해요!